cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -272,6 +329,8 @@ public void drop(Id.DatasetInstance instance) throws Exception {
     LOG.info("Deleting dataset {}.{}", instance.getNamespaceId(), instance.getId());
     dropDataset(instance, spec);
     publishAudit(instance, AuditType.DELETE);
+    // revoke privileges
+    authorizerInstantiator.get().revoke(instance.toEntityId());
```

Comment:
what does this do? revoke rights for all users? That means if I create a dataset, give access to a group of people, then for some reason I need to drop and recreate the dataset, I have to grant again? Is that by design?